### DISCUSSION TITLE
What AI Models are you interested to try out from the New GitHub Models Catalog? Share Your Thoughts!

### DISCUSSION THREAD

**Discussion by samus-aran** [community/community#134404] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

We're excited to announce the launch of [GitHub Models](https://github.blog/news-insights/product-news/introducing-github-models/), including the all [new catalog](https://github.com/marketplace/models)! 🎉 This catalog features a diverse range of models ranging from model types, sizes, and specializations. Now that the catalog is live, we want to hear from you!

<img width="1364" alt="Screenshot 2024-08-01 at 16 47 48" src="https://github.com/user-attachments/assets/2b5903fd-b8cc-4b99-9db3-7a60cebc8ca4">

>Alt Text: The graphic is a screenshot of the Models page on GitHub Marketplace. It features a number of items with various AI model names listed.

## What's your favorite AI model from the new catalog and why?

* Which model caught your eye and why?
* Have you had a chance to experiment with any of the new models? How was your experience?
* Is there a particular model that you hope for in the future to be included?
* Are there any emerging models that you're excited about?

Whether you're into machine learning, deep learning, natural language processing, computer vision, or any other AI domain, your feedback is invaluable. Your insights can help others discover powerful tools and spark innovative ideas.

We believe every developer can be an AI engineer with the right tools and training. From playground to coding with the model in GitHub Codespaces to production deployment via Azure, GitHub Models shows you how simple it can be. Sign up for the limited public beta [HERE](https://gh.io/models). 

**Share your experiences and feedback in the comments below or create a new discussion 👇**

**Comment by potatoqualitee** [comment]
### Which model caught your eye and why?

Llama 403B - I haven't had a chance to use it elsewhere and can't deploy it at home.

### Have you had a chance to experiment with any of the new models? How was your experience?

Yes. It's an awesome concept and I'm excited about this. Encountered one bug so far: I can't modify the tokens by using my mouse. I can only modify it by typing in a number, for example 4096.

![image](https://github.com/user-attachments/assets/b69d039c-1e94-4227-80f8-1bc9fbc0b89c)

### Is there a particular model that you hope for in the future to be included?

Any hyped model (like Llama 403B in this case), Claude

### Are there any emerging models that you're excited about?

Flux

**Comment by squew** [comment]
I would say at this point, it is best to learn practical like agentic, function calling, rag, etc. than trying specific models. Of course, new models mean new strategies, new possibilities, so both are linked. etc.
That said, while the smaller versions of some models can run on self-hosted, having Mistral 123B or a full version of Cohere R+, llama3.1, in fact all the current models in the marketplace are interesting... and it would take some time to evaluate them. Future models could be more capable of understanding us (even nonverbal?), and domains like biosciences or weather forecast (that are some models interesting to provide), have the potential to have a significant impact. Other models will keep growing and complexifying and have specific areas for different purposes (MoE or the brain are examples), but sync and communication is hard so maybe agents will prevail...

 
For the future, having models that can understand a computer screen and eventually help the user (step toward and always on assistant), and models like Segment Anything 2 would be great. Rather than big models like Stable Diffusion, smaller, task-specific models lighter and simpler could be interesting. A workflow (altering the color of a t-short), segmentation + labeling (which is ‘wow’), with a mask as an output used to apply changes to the original picture. Each task can be isolated.

The app needs to know what the user wants, in natural language. So, some NLP and an occasion to use the semantic kernel to interpret the intent, two LLM agents (or SLM) could be doing the same job (or having the same goal, but using different strategies, like clip, or segmenting already), then a third one would summarize and validate before proceeding. 
It uses agents (maybe Autogen), computer vision and image, function calling, NLP — all integrated in one workflow, or can be separated. So, mostly tools I’d like to learn ☺️

For NLP, a simple case is an Assistant like Google’s (but smarter), that can tell if the user is just chatting or asking something (like turn on that light). Available models on the marketplace, are capable of this, but with some help.
Another topic: analyzing a text and rate its level of "exaggeration" true if there are many overstatements, or false (with limited fact checking).
Fictional example, if words like revolutionary, groundbreaking (fact checking has existed for a decade) and constant use of laudatory vocabulary > true), it's a step, towards rating news, scientific papers, any text in fact. So, it might use smaller more specialized models, but which are especially useful.
A workflow to collect CVE, with the usual info, but also select a clear perimeter: kernel, system components, or a rogue sticker cannot be compared (hard to find that for free), and would use a vector to follow up. 

Finally, time series models, to detect anomalies and thresholds. Example, the number of card payments/min (I work in that field) has a regular pattern, except on special days (not regularly occurring), national holiday, or black Friday are quite different, so irregular seasonality. This is a frequent problem. 

And all of this in one click with Codespaces :)

These are just ideas without name of models because I haven’t been following closely new models… but application of the models, are in my opinion, more important now.

**Comment by Ammar-Alnagar** [comment]
Definitely the mamba models

**Comment by aluxian** [comment]
The top of the line models are a must: GPT-4, Claude 3.5 Sonnet, Claude Opus. Also coding specific ones: DeepSeek-V2

**Comment by lipeilin375** [comment]
A new catalog is a truly nice place to try out the models at one place, but by the way, i want to share a small feedback that a lot of AI engineers might also have, that how can we upload our own models to the community and in facts, i think there can be another type of repository that could allow us to upload the model file, train file and so on. This might promote open-source community's improvement~

hope to be seen~

**Comment by mgagvani** [comment]
Vision transformer models such as Microsoft Florence, ViT, etc. I think would be a good addition. Obviously we have some multimodal models in there already but I think vision specific models would be a good addition.

**Comment by NazimHAli** [comment]
Mamba models
MoE models

Most importantly, allow developers upload their own.

**Comment by raghusbhat** [comment]
VLLMs (vision) and GAN models please!
Phi-3.5-vision
Qwen2-VL
Florence-2
Also, in playground, options to finetune, prune, transfer learn, model conversion etc would be really awesome.

**Comment by Kjdragan** [comment]
Getting access to ColPali would be great.  Looks like the future of RAG through vision models.

**Comment by bella8866** [comment]
I’m especially interested in exploring models focused on natural language processing (NLP), such as those based on GPT, as well as computer vision models for image recognition, because these are my requirements as I work as an  SEO Specialist and content writer for site [ihatecbts](https://ihatecbts.net/)

**Comment by curiosityz** [comment]
I'd like to be able to train or finetine models, too. I'm surprised by how good the Llama 403B is!

**Comment by Shamik-07** [comment]
How about some diffusion models such as Flux?

**Comment by timnity** [comment]
DeepSeek V2.5 of cause

**Reply by Nicat-dcw** [reply]
and deepseek r1

**Comment by sezginsevinc** [comment]
Is it possible Flux.1 Pro?

**Comment by Freakyty** [comment]
No

**Comment by Freakyty** [comment]
Maybe

**Comment by crazygo** [comment]
Hope to launch a vertical model specifically for Review or Unit Test.

**Comment by fcore24076** [comment]
i would like to have claude 3.5 sonnet new

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by Nicat-dcw** [comment]
I would like to use antrophic's claude serie models like opus,sonnet and haiku

**Comment by yaph** [comment]
I added support for GitHub Models to my terminal based chat client [Charla](https://github.com/yaph/charla) after I got access to the public beta 2 months ago. I use the Azure AI Inference Python library for interacting with GitHub models and the Ollama client and server for local models.

For simple tasks, small models like Phi3 work fine in my experience. For more complex tasks, a model like Llama3 can be a good choice, but none come close to the gpt-4o models available through GitHub Models, in my opinion. I'm amazed at the speed of the responses, the stability of the API, and, most importantly, the quality of the output. Fun fact: the responses from the remote API are often faster than interactions with local models.

Regarding the code libraries I'm using, Ollama offers some features that I miss in the Azure API, for example, metadata about the models via the `show` function. More importantly, to maintain the chat history in Ollama, you can access the context as a list of integers from each chat completion response and pass it back in your next request. This makes it quite easy to limit the context size compared to working with lists of different message-type objects in Azure. I also assume it uses less bandwidth, though I haven’t tested this.

Anyhow, I really like how easy it was to integrate GitHub Models and how reliably the API works. I’m looking forward to using more specialized and smaller models in the future that are even faster and more efficient.

**Comment by greghouston** [comment]
DeepSeek V3

**Comment by wisteriesDev** [comment]
New Mistral Small o/
